rdoxygen-design.md

```diff
@@ -0,0 +1,126 @@
+
+    An obvious location for this file is `inst/doc/doxygen`.  However, `inst/doc` is typically `.gitignore`d (e.g., by `devtools::use_vignette()` and `usethis::use_vignettes()`).  So we'd have to manually exclude `inst/doc/doxygen/Doxyfile`, via appending an existing `.gitignore` with something like [this](https://stackoverflow.com/questions/5533050/gitignore-exclude-folder-but-include-specific-subfolder):
+    
+    ```
+    # paste the following to the bottom of existing .gitignore
+    
+    !inst/doc # unignores inst/doc
+    inst/doc/* # ignore everything inside inst/doc but not inst/doc itself
+    !inst/doc/doxygen/Doxyfile # unignore Doxyfile
+    ```
+    
+    Another option is to have `Doxyfile` live in `vignettes`.  If creating Doxgygen vignettes is always enabled (i.e., not optional), then this folder already exists, so easier to put `Doxyfile` here than to edit the `.gitignore` as above.  However, if the folder doesn't exist, then perhaps it's confusing to create the folder just to store the `Doxyfile`.
+    
+    Yet another option is to ship a list of key-value pairs for Doxygen tags instead of the `Doxyfile` itself, e.g., something like
+    
+    ```r
+    # put this in any file in R/ subfolder
+    .doxy_opts <- c(AUTOLINK_SUPPORT = "NO")
+    ```
+    
+    As a package developer, this seems like an easier way to view package-dependent options (rather than scrolling through the `Doxyfile`).  However, the **rdoxygen** package is then needed to parse these options, and I would rather a package I'm authoring which provides Doxygen documentation not require all users to install **rdoxygen** as well.  Perhaps the `.R` file containing the lines above could be `.Rbuildignore`d but not `.gitignore`d?
+   
+## Default Doxyfile Tags
+
+First, it should be noted that the `INPUT` tag cannot handle relative directories outside of where it's run, i.e.:
+
+```bash
+# in Doxyfile
+INPUT = src/ # works fine
+INPUT = ../src/ # does not work
+```
+
+Thus, **rdoxygen** should run Doxygen *from the package root folder*.  That being said, here are the default Tags I'm suggesting:
+
+```bash
+INPUT = src/ inst/include/ # the two locations in which you expect to find C++ code
+OUTPUT_DIRECTORY = inst/doc/doxygen
+PROJECT_NAME = "C++ Library Documentation for Package PackageName"
+GENERATE_LATEX = NO # probably not needed in most cases?
+```
+
+Note that I did not suggest `EXTRACT_ALL = YES`, because I wonder whether inexperienced Doxygen users might think they need to document every single function for their Doxygen output to work properly...
+
+Also note that another useful option might be `USE_MATHJAX = YES`, which makes formulas look much nicer than when this option is set to `NO`...
```

Comment:
Does this come at the cost of more dependencies?

Reply:
Usually not.  Mathjax is rendered on-the-fly, typically via a public [CDN](http://docs.mathjax.org/en/latest/configuration.html), which means you need an internet connection for the equations in the Doxy-doc to display properly.  It's also possible to install mathjax locally: then you don't need the internet to display the math correctly, at the cost of the additional dependency.  

Besides having much uglier equations, another downside to not using Mathjax is that each equation is then shipped as its own separate png, which makes the size of the package much larger.  I believe that CRAN is strongly encouraging package submissions of minimal size.  Moreover, essentially all rmarkdown vignettes use Mathjax via CDN (hosted by RStudio)...

In the balance, my opinion is that the price to pay for defaulting to Mathjax is rather minimal.  Besides, it would be simple for the user to disable, and we could provide roxygen `@details` to clarify...